hrms/hr/doctype/attendance_request/attendance_request.py

```diff
@@ -19,7 +19,6 @@ class OverlappingAttendanceRequestError(frappe.ValidationError):
 class AttendanceRequest(Document):
 	def validate(self):
 		validate_active_employee(self.employee)
-		validate_dates(self, self.from_date, self.to_date)
```

Comment:
This validation checks if `to_date` is less than `from_date`
Attendance Request does not have an explicit validation for future dates, I think. It's only in the Attendance doctype

Reply:
okay, you want to say that we have to add a new custom validation to the attendance request and simply remove the "future dates are not valid" validation. right?

new validation like
```py
def validate_dates_for_ar(doc, from_date, to_date):
    date_of_joining, relieving_date = frappe.db.get_value(
        "Employee", doc.employee, ["date_of_joining", "relieving_date"]
    )
    if getdate(from_date) > getdate(to_date):
        frappe.throw(_("To date cannot be less than from date"))
    elif date_of_joining and getdate(from_date) < getdate(date_of_joining):
        frappe.throw(_("From date cannot be less than employee's joining date"))
    elif relieving_date and getdate(to_date) > getdate(relieving_date):
        frappe.throw(_("To date cannot be greater than employee's relieving date"))
```